packages/neo-one-editor/src/manager/PackageManager.ts

```diff
@@ -67,46 +103,157 @@ export class PackageManager {
+      async (filePath) =>
+        // tslint:disable-next-line:promise-must-complete
+        new Promise<string>((resolve, reject) => {
+          this.mutableQueue.push({
+            name,
+            version,
+            filePath,
+            resolve,
+            reject,
+          });
+        }),
+    );
+    const zippedPackageInfo = _.zip(packageFilePaths, packageFilePromises) as ReadonlyArray<[string, Promise<string>]>;
+
+    this.startFetchLoop().catch(() => {
+      // do nothing
+    });
+
+    const packageFiles = await Promise.all(
+      zippedPackageInfo.map(async ([filePath, filePromise]) => {
+        const file = await filePromise;
+
+        return {
+          file,
+          path: filePath,
+        };
+      }),
+    );
+
     return {
-      name: depInfo.name,
-      fileName: file === undefined ? 'index.js' : file,
-      version: depInfo.version,
-      dependency: pkg,
-      packageJSON: pkgJSON,
-      types,
-      typeFile: depInfo.types === undefined ? 'index.d.ts' : depInfo.types,
-      subDependencies: subPkgs,
+      name,
+      version,
+      packageFiles,
+      subDependencies,
     };
   }
 
-  private async downloadDependency({ name, version, file }: FetchPackageInfo) {
+  private async getAllPackageFiles({ name, version }: FetchPackageInfo): Promise<ReadonlyArray<string>> {
+    const res = await fetch(this.getNPMDataUrl({ name, version }));
+    const fileTree = await res.json();
+
+    return this.getDirectoryList({ dirList: fileTree.files });
+  }
+
+  private getDirectoryList({
+    dirList,
+    currentPath,
+  }: {
+    readonly dirList: ReadonlyArray<FileData | DirectoryData>;
+    readonly currentPath?: string;
+  }): ReadonlyArray<string> {
+    return _.flatten(
+      dirList.map((entry) => {
+        if (entry.type === 'directory') {
+          return this.getDirectoryList({
+            dirList: entry.files,
+            currentPath: currentPath === undefined ? entry.name : `${currentPath}/${entry.name}`,
+          });
+        }
+
+        return currentPath === undefined ? entry.name : `${currentPath}/${entry.name}`;
+      }),
+    );
+  }
+
+  private async checkTypes({ name, version, types }: FetchPackageInfo) {
+    if (types) {
+      return;
+    }
+
+    const typesName = `@types/${name.replace('@', '').replace('/', '__')}`;
+    const versionRes = await fetch(this.getNPMDataUrl({ name: typesName }));
+    if (!versionRes.ok) {
+      return;
+    }
+    const typesVersions = await versionRes.json();
+    const latestTypesVersion = typesVersions.versions[0];
+
+    const res = await fetch(this.getNPMDataUrl({ name: typesName, version: latestTypesVersion }));
+    if (!res.ok) {
+      return;
+    }
+    const files = await res.json();
+    let indexFound = false;
+    files.files.forEach((file: FileData | DirectoryData) => {
+      if (file.name === 'index.d.ts') {
+        indexFound = true;
+      }
+    });
+    if (!indexFound) {
```

Comment:
You don't need to find an `index.d.ts` file for it to be a valid `@types`. As long as the package exists, just invoke the callback.